lib/internal/process/promises.js

```diff
@@ -29,6 +51,7 @@ function resolveError(type, promise, reason) {
   process.nextTick(() => {
     process.emit('multipleResolves', type, promise, reason);
   });
+  setHasRejectionToWarn(false);
```

Comment:
Is this required? Without looking at it further, it does seem to be somewhat independent.

Reply:
I remember removing this caused tests to fail which was what prompted me to make this falsy value explicit in the first place.

Reply:
That sounds somewhat weird to me. Could you have another look into it? I just checked the new implementation again and I believe `setHasRejectionToWarn(false)` should only be called https://github.com/nodejs/node/pull/25200/files#diff-e7ef4821107f4cae3bd0fea4dec350bfR83 here (so https://github.com/nodejs/node/pull/25200/files#diff-abbefee1c196cd57720161c414eaaa96R84 also seems wrong to me but I did not check the code and just read checked the code changes).

Reply:
Looks like these are not necessary after all. I've removed those.